tests/std/tests/GH_004597_self_swap/test.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <utility>
+
+struct swap_counter {
+    unsigned int* pcnt_ = nullptr;
+
+    friend _CONSTEXPR20 void swap(swap_counter& lhs, swap_counter& rhs) noexcept {
+        std::swap(lhs.pcnt_, rhs.pcnt_);
+        if (lhs.pcnt_ != nullptr) {
+            ++(*lhs.pcnt_);
+        }
+        if (rhs.pcnt_ != nullptr) {
+            ++(*rhs.pcnt_);
+        }
+    }
+
+    _CONSTEXPR20 bool operator==(unsigned int x) const {
+        return *pcnt_ == x;
+    }
+};
+
+
+_CONSTEXPR20 bool test_gh_4595() {

```

Comment:
4595 isn't the right bug number - it should be #4597, as in the test directory name. We should also have a comment reminding us what the bug title was.

Reply:
I remove it beacuse @frederick-vs-ja said:
> This line is redundant because GH_004597 is already in the folder name.

So we still need it.